### ISSUE TITLE
PrintToConsoleAll

### ISSUE THREAD

**Issue by phorcep** [Rainyan/sourcemod-nt-anti-cap-deny#1]
PrintToConsoleAll was not added to SourceMod until 1.9 which prevents this from compiling under lower versions (specifically attempted in 1.8.0).

**Comment by Rainyan** [comment]
Thanks for reporting.

I pushed updates that should fix this for SM 1.8, could you confirm if it's working for you now?

**Comment by phorcep** [comment]
`nt_anti_ghostcap_deny.sp(283) : error 092: number of arguments does not match definition`

**Comment by Rainyan** [comment]
Are you sure you're compiling with the [latest update](https://github.com/Rainyan/sourcemod-nt-anti-cap-deny/archive/refs/heads/master.zip) (`PLUGIN_VERSION` should read 1.1.2 on [line 7 of the .sp file](https://github.com/Rainyan/sourcemod-nt-anti-cap-deny/blob/4cd38620cddf72c7a12fd7405fc2f05e082ef3dc/scripting/nt_anti_ghostcap_deny.sp#L6-L8)). I had to push several updates due to missing some bits at first.

**Comment by phorcep** [comment]
Yes, 1.1.2

**Comment by Rainyan** [comment]
Edit: This should hopefully be fixed now, check the last comment.

- - -

~I'm unable to reproduce this on [SourceMod 1.8 branch latest](https://sm.alliedmods.net/smdrop/1.8/) _(sourcemod-1.8.0-git6050)_, with both Windows `spcomp.exe` and on the Linux `spcomp`.~

~Are you trying to compile on Linux (Wine), with Windows spcomp executable? It could be related to your setup somehow, if so, although I'm not sure how. I don't have the wineconsole stuff mentioned in your Linux SRCDS guide set up, so I didn't test the compile with that.~

Here's a test script I was using on Linux for downloading & compiling the Linux branch, if it's of any use with debugging:
```sh
#!/usr/bin/sh

# Which SourceMod branch to download?
SM_BRANCH="1.8"

# Which platform (linux/windows/mac)
SM_PLATFORM="linux"

# Build the download url for latest release on this branch/platform
BASE_URL="https://sm.alliedmods.net/smdrop"
LATEST_OF_BRANCH="`wget -qO - ${BASE_URL}/${SM_BRANCH}/sourcemod-latest-${SM_PLATFORM}`"
echo "Latest ${SM_PLATFORM} release in the ${SM_BRANCH} branch is: ${LATEST_OF_BRANCH}"
# Generate the full SM archive download url
DOWNLOAD_URL="${BASE_URL}/${SM_BRANCH}/${LATEST_OF_BRANCH}"

# Get the file...
wget --no-clobber $DOWNLOAD_URL -P "./${SM_PLATFORM}"
# ...and unarchive it
cd $SM_PLATFORM
  if [ $SM_PLATFORM == "linux" ]; then
    tar -xvzf "${LATEST_OF_BRANCH}"
  else
    unzip -n "${LATEST_OF_BRANCH}"
  fi
cd ..

# Get the plugin files
SCRIPTING_PATH="$(pwd)/${SM_PLATFORM}/addons/sourcemod/scripting"
# Install the neotokyo.inc SM include
wget "https://raw.githubusercontent.com/softashell/sourcemod-nt-include/master/scripting/include/neotokyo.inc" -O "$SCRIPTING_PATH/include/neotokyo.inc"
# Download the plugin code
wget "https://raw.githubusercontent.com/Rainyan/sourcemod-nt-anti-cap-deny/master/scripting/nt_anti_ghostcap_deny.sp" -O "${SCRIPTING_PATH}/nt_anti_ghostcap_deny.sp"

# Compile
if [ $SM_PLATFORM == "linux" ] || [ $SM_PLATFORM == "mac" ]; then
  ${SCRIPTING_PATH}/spcomp "${SCRIPTING_PATH}/nt_anti_ghostcap_deny.sp"
else
  # Run with Wine(???)
  ${SCRIPTING_PATH}/spcomp.exe "${SCRIPTING_PATH}/nt_anti_ghostcap_deny.sp"
fi
# Compile successful?
echo -e "\nResult(s): $(ls |grep smx)"
```

TL;DR: Perhaps try with a newer (newest?) 1.8 branch release, if upgrading to latest stable isn't feasible?

**Comment by Rainyan** [comment]
[1.8 build 5992](https://www.sourcemod.net/downloads.php?branch=1.8-dev&all=1) makes reference to the error:

> Add missing parameter to DataPack.IsReadable (#603)

Perhaps you should try a 1.8 release >= this build version?

I can also probably add an exception for the pre 1.8-5992 releases, if you're not in a hurry and don't wanna bother updating.

**Comment by Rainyan** [comment]
Ok, I'm pretty sure it's a bug in 1.8 builds in range 5535 - 5991.

I pushed an update that should make this compile on all 1.8 branch releases now (this problematic range included), so you probably don't need to update the server's SourceMod after all. Fingers crossed.

**Comment by phorcep** [comment]
Works now, thanks.
